projects/cdk/components/dialog-host/dialog-host.component.ts

```diff
@@ -33,22 +43,42 @@ const isFakeHistoryState = (
     animations: [TUI_PARENT_ANIMATION],
 })
 export class TuiDialogHostComponent<T extends TuiDialog<unknown, unknown>> {
-    readonly dialogs$ = combineLatest(this.dialogsByType).pipe(
-        map(allTypesDialogs =>
-            new Array<T>()
-                .concat(...allTypesDialogs)
-                .sort((a, b) => a.createdAt - b.createdAt),
-        ),
-    );
+    public dialogs: readonly T[] = [];
+    public isDialogClosesOnBack = defaultClosesOnBack;
 
     constructor(
         @Inject(TUI_DIALOG_CLOSES_ON_BACK)
-        readonly isDialogClosesOnBack$: Observable<boolean>,
+        isDialogClosesOnBack$: Observable<boolean>,
         @Inject(TUI_DIALOGS)
-        private readonly dialogsByType: Array<Observable<readonly T[]>>,
+        dialogsByType: Array<Observable<readonly T[]>>,
         @Inject(HISTORY) private readonly historyRef: History,
         @Inject(Title) private readonly titleService: Title,
-    ) {}
+        @Self() @Inject(TuiDestroyService) destroy$: Observable<void>,
+        @Inject(ChangeDetectorRef) cdr: ChangeDetectorRef,
+    ) {
+        // Due to this view being parallel to app content, `markForCheck` from `async` pipe
+        // can happen after view was checked, so calling `detectChanges` instead
+        combineLatest(dialogsByType)
+            .pipe(
+                map(arr =>
+                    new Array<T>()
```

Comment:
What's the point of `new Array<T>().concat`? Why not just `[...arr]`?

Reply:
`[...arr]` wouldn't flatten the array properly. I had originally used `arr.flat()` however at build time it threw an error because it's only targeting ES2019.